pkg/cli/config/adminkubeconfig/new_admin_kubeconfig.go

```diff
@@ -0,0 +1,267 @@
+		IOStreams:        streams,
+	}
+}
+
+func NewCmdNewAdminKubeconfigOptions(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewNewAdminKubeconfigOptions(restClientGetter, streams)
+
+	cmd := &cobra.Command{
+		Use:                   "new-admin-kubeconfig",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Generate, make the server trust, and display a new admin.kubeconfig."),
+		Long:                  newAdminKubeconfigLong,
+		Example:               newAdminKubeconfigExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete())
+			cmdutil.CheckErr(o.Validate(args))
+			r, err := o.ToRuntime()
+			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(r.Run(context.TODO()))
+
+		},
+	}
+
+	o.AddFlags(cmd)
+
+	return cmd
+}
+
+func (o *NewAdminKubeconfigOptions) AddFlags(cmd *cobra.Command) {
+}
+
+func (o *NewAdminKubeconfigOptions) Complete() error {
+
+	return nil
+}
+
+func (o NewAdminKubeconfigOptions) Validate(args []string) error {
+	if len(args) > 0 {
+		return fmt.Errorf("no arguments allowed")
+	}
+
+	return nil
+}
+
+func (o *NewAdminKubeconfigOptions) ToRuntime() (*NewAdminKubeconfigRuntime, error) {
+	clientConfig, err := o.RESTClientGetter.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	kubeClient, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := &NewAdminKubeconfigRuntime{
+		KubeClient: kubeClient,
+		Host:       clientConfig.Host,
+		IOStreams:  o.IOStreams,
+	}
+
+	return ret, nil
+}
+
+type NewAdminKubeconfigRuntime struct {
+	KubeClient kubernetes.Interface
+	Host       string
+
+	genericclioptions.IOStreams
+}
+
+func (r *NewAdminKubeconfigRuntime) Run(ctx context.Context) error {
+	systemMasterClientCert, signerCertBytes, err := createNewAdminClientCert()
+	if err != nil {
+		return fmt.Errorf("unable to create new client certificate: %w", err)
+	}
+
+	serverCABundle, err := refreshcabundle.GetCABundleToTrustKubeAPIServer(ctx, r.KubeClient)
+	if err != nil {
+		return fmt.Errorf("unable to get the CA bundle from the cluster: %w", err)
+	}
+
+	// update the in-cluster configmap
+	existingConfigMap, err := r.KubeClient.CoreV1().ConfigMaps("openshift-config").Get(ctx, adminKubeconfigClientCAConfigMap, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		existingConfigMap, err = r.KubeClient.CoreV1().ConfigMaps("openshift-config").Create(ctx, &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "openshift-config",
+				Name:      adminKubeconfigClientCAConfigMap,
+			},
+		}, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to create starting --namespace=openshift-config configmap/%s : %w", adminKubeconfigClientCAConfigMap, err)
+		}
+	} else if err != nil {
+		return fmt.Errorf("unable to read configmap %w", err)
+	}
+	caBundle, err := combineCABundles(existingConfigMap.Data["ca-bundle.crt"], string(signerCertBytes))
```

Comment:
This is ensuring we continue trust the old keys for now?  Worth a comment if so.